getconf/finders.py

```diff
@@ -111,7 +111,7 @@ def __init__(self, config_files):
             if path is None:
                 continue
             # Handle '~/.foobar.conf'
-            path = os.path.abspath(os.path.expanduser(path))
+            path = os.path.abspath(os.path.expanduser(str(path)))
```

Comment:
Since we're adding support for `pathlib.Path` here, shouldn't we convert `path` to a `pathlib.Path` first, and use its methods, instead?

Reply:
Also, we should add a comment explaining the different types we're handling here, and in the docs for `config_files=`.

Reply:
As I said in my commit message, it's another job. I started by doing this but it was too much work, and not even related to path support.

I agree this added support should be mentioned in the docs, though.

Reply:
`config_files` docs should be updated indeed. The internal refactor to use `pathlib.Path` internally could certainly happen in an other PR.